operators/online_video_reasoner/online_video_reasoner.py

```
@@ -0,0 +1,547 @@
+_RESERVED_REQUEST_OPTIONS = {"messages", "model", "stream"}
+
+
+def _validate_endpoint(endpoint: str) -> None:
+    """Require HTTPS except for an explicitly local cleartext endpoint."""
+    if not isinstance(endpoint, str):
+        raise ValueError("endpoint must be a valid HTTP or HTTPS URL")
+    try:
+        parsed = urlsplit(endpoint)
+        hostname = parsed.hostname
+        # Accessing port validates both its syntax and range.
+        parsed.port
```

Ruff (0.15.21)

[warning] Found useless expression. Either assign it to a variable or remove it.

(B018)